core/runtime/src/main/java/io/quarkus/runtime/configuration/NativeConfigBuilder.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.runtime.configuration;
+
+import static io.smallrye.config.PropertiesConfigSourceLoader.inFileSystem;
+
+import java.nio.file.Paths;
+
+import io.smallrye.config.SmallRyeConfigBuilder;
+
+/**
+ * Do not register classpath resources lookup in native mode to avoid missing resources registration errors, which
+ * became a strict check during native image execution.
+ * <p>
+ * Configuration coming from classpath resources is recoded during build time in a low ordinal source, so the
+ * configuration will still be available. If the users change the ordinals of the sources in runtime, it may
+ * cause unexpected values to be returned by the config, but this has always been the case. The classpath configuration
+ * resources were never registered in the native image.
```

Comment:
Is this true? 

In https://github.com/oracle/graal/discussions/10264#discussioncomment-11604091 we describe a scenario were using the classpath would work fine (assuming the user registers the resource for reflection and sets the ordinals accordingly).

Reply:
With our stuff yes, we never registered the resources. The configuration is just included in a class file and we discard the resources, which may cause the issue described in https://github.com/quarkusio/quarkus/pull/44652#issuecomment-2497713122, but this has been an issue since day one and until now we never got any complaints about it.